T34138 stop restart kolibri

manuq opened the pull request:
https://phabricator.endlessm.com/T34138

manuq commented:
Please don't merge until the refactor at #78 is merged.

dylanmccall commented on .p4a:
```diff
@@ -11,7 +11,6 @@
 --allow-minsdk-ndkapi-mismatch
 --permission ACCESS_NETWORK_STATE
 --permission FOREGROUND_SERVICE
---service workers:workers.py
```
:+1: This should remove some uncertainty, at least.

manuq committed:
Run Kolibri service workers in main activity

Remove the foreground service which was never shutting down properly.
Instead import ServicesPlugin from Kolibri and make it a listener of the
process bus. So that when the process bus stops the workers shutdown
properly.

https://phabricator.endlessm.com/T34138

manuq committed:
Lifecycle: Stop Kolibri when main activity stops

And run Kolibri again when the main activity is resumed.

This basically hooks the Kolibri process lifecycle with Android's main
activity lifecycle.

https://phabricator.endlessm.com/T34138

dylanmccall commented on src/kolibri_android/main_activity/activity.py:
```diff
@@ -94,6 +94,16 @@ def __init__(self):
+    def on_activity_stopped(self, activity):
+        super().on_activity_stopped(activity)
+        if self._kolibri_bus is not None:
+            self._kolibri_bus.transition("STOP")
```
I was a little suspicious of this since it's transitioning to different target states than we use [in kolibri-daemon](https://github.com/learningequality/kolibri-installer-gnome/blob/master/src/kolibri_daemon/kolibri_http_process.py#L103-L122) and I remember that being troublesome, but it all checks out. I wonder if these state transitions were improved with 0.16? (Or I could just be remembering something else).

manuq replied:
Oh interesting. I didn't remember that you had to play with magicbus in kolibri-daemon as well. I will test what happens if change it to transition to the "IDLE" state.

Also in the playthrough yesterday we saw that the contentimport task doesn't obey to cancel as fast as channelimport. It could go for several seconds. So I think I'll need to add a waiter similar to what Kolibri 16 has around here https://github.com/learningequality/kolibri/blob/ecf08f1f07b3ae8c6597ccaf708698c8e918e032/kolibri/utils/server.py#L599.

dylanmccall reviewed (approved):
Oops, we both rebased this branch this morning. I should have let you know on Slack after I force-pushed it. (I wish git had a "force push but only if I was the last person to touch the remote" button). But don't worry about the mysterious pushes from me that github mentions. Those were the exact same thing you did, including dropping that change to `get_home_folder()`.

Anyway, I like it! Nice and simple :)

manuq commented:
> Oops, we both rebased this branch this morning. I should have let you know on Slack after I force-pushed it. (I wish git had a "force push but only if I was the last person to touch the remote" button). But don't worry about the mysterious pushes from me that github mentions. Those were the exact same thing you did, including dropping that change to `get_home_folder()`.

Ah, thanks for doing so and sorry for messing up! I didn't look at github and just force-pushed from my end.
